# HG changeset patch
# User Xerxes Ranby <xerxes@zafena.se>
# Date 1314700204 -7200
# Node ID 5e706bdbcce960b39cd58663d2bc4e8afddcfd62
# Parent  82093e560de226787b4c4dca51ad95aaa73a3482
CACAO: CA149: Used wrong class loader, Updated to 2011-08-29 revision.

2011-08-30  Xerxes Ranby  <xerxes@zafena.se>

	CACAO
	- CA149: Used wrong class loader.
	- src/vm/javaobjects.cpp (java_lang_reflect_Method::invoke):
	  [OPENJDK] stack index of caller was off by one, causing many apt
	  (Annotation Processing Tool) failures.
	* NEWS: Updated.
	* Makefile.am
	(CACAO_VERSION): Updated CACAO to 2011-08-29 revision.
	(CACAO_SHA256SUM): Updated.

diff -r 82093e560de2 -r 5e706bdbcce9 ChangeLog
--- a/ChangeLog	Tue Aug 09 11:34:28 2011 +0200
+++ b/ChangeLog	Tue Aug 30 12:30:04 2011 +0200
@@ -1,3 +1,15 @@
+2011-08-30  Xerxes Rånby  <xerxes@zafena.se>
+
+	CACAO
+	- CA149: Used wrong class loader.
+	- src/vm/javaobjects.cpp (java_lang_reflect_Method::invoke):
+	  [OPENJDK] stack index of caller was off by one, causing many apt
+	  (Annotation Processing Tool) failures.
+	* NEWS: Updated.
+	* Makefile.am
+	(CACAO_VERSION): Updated CACAO to 2011-08-29 revision.
+	(CACAO_SHA256SUM): Updated.
+
 2011-08-09  Xerxes Rånby  <xerxes@zafena.se>
 
 	CACAO
diff -r 82093e560de2 -r 5e706bdbcce9 Makefile.am
--- a/Makefile.am	Tue Aug 09 11:34:28 2011 +0200
+++ b/Makefile.am	Tue Aug 30 12:30:04 2011 +0200
@@ -18,8 +18,8 @@
 LANGTOOLS_SHA256SUM = 9ddc00ec50fd2f5e331dc2bc10da4e23b69bf644eb92d50b39a2003c18fb5aa1
 OPENJDK_SHA256SUM = 4043a75c2c4385dd735f8dbbf2369311ce1b951217c9dbe9bba9609e24eb291e
 
-CACAO_VERSION = 12d5cd5d2dfa
-CACAO_SHA256SUM = f4a93710709f43e203d79c1c8bb73f14ebcca41be75887fb92308b86747312c5
+CACAO_VERSION = 4549072ab2de
+CACAO_SHA256SUM = ec097ce96f047abbaeab4faab752c121d29f08bdeae358038a0b7cde03361d07
 CACAO_BASE_URL = http://icedtea.classpath.org/download/drops/cacao
 CACAO_URL = $(CACAO_BASE_URL)/$(CACAO_VERSION).tar.gz
 CACAO_SRC_ZIP = cacao-$(CACAO_VERSION).tar.bz2
diff -r 82093e560de2 -r 5e706bdbcce9 NEWS
--- a/NEWS	Tue Aug 09 11:34:28 2011 +0200
+++ b/NEWS	Tue Aug 30 12:30:04 2011 +0200
@@ -107,7 +107,11 @@
   - MIPS: 64-bit and interpreter inlining by default.
   - Trivial implementation of stubs for MIPS.
 * CACAO
+  - CA149: Used wrong class loader.
   - CA159: Exception handler blocks / register mixup.
+  - src/vm/javaobjects.cpp (java_lang_reflect_Method::invoke): [OPENJDK] stack
+    index of caller was off by one, causing many apt
+    (Annotation Processing Tool) failures.
   - Set thread to RUNNABLE during Thread.start.
   - Removed state-setting function call that would be done by the thread itself,
     creating a nasty race.
